src/exactio/exactio_exanic.c

```diff
@@ -231,15 +231,24 @@ static int set_exanic_params(exanic_t *exanic, char* device, int port_number,
         exit(1);
     }
 
+    int ifr_changed = 0;
+
     if(promisc)
+    {
+        ifr_changed |= !(ifr.ifr_flags & IFF_PROMISC);
         ifr.ifr_flags |= IFF_PROMISC;
+    }
 
+    ifr_changed |= !(ifr.ifr_flags & IFF_UP);
     ifr.ifr_flags |= IFF_UP;
 
-    if (ioctl(fd, SIOCSIFFLAGS, &ifr) == -1)
+    if (ifr_changed)
     {
-        fprintf(stderr, "ioctl(SIOCSIFFLAGS): %s:%d: %s\n", device, port_number, strerror(errno));
-        exit(1);
+        if (ioctl(fd, SIOCSIFFLAGS, &ifr) == -1)
+        {
+            fprintf(stderr, "ioctl(SIOCSIFFLAGS): %s:%d: %s\n", device, port_number, strerror(errno));
+            exit(1);
```

Comment:
Be careful with a simple exit here. There's a whole chain of dependencies that lead us to this function, so we're probably deep inside a thread construction block. Can I suggest a return statement here. It should cascade upwards correctly.

Reply:
There is already an exit in this function which I did not add.
Do you want all of them to be changed to return 1?